Smithy fails to build on OSX

Hello

I'm trying to build Smithy on OSX 11.2.3, but it fails with the following error:

```
/Users/arve/Projects/grafana/smithy/smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelAssembler.java:481: error: unexpected heading used: <H2>, compared to implicit preceding heading: <H3>
     * <h2>Implementation notes</h2>
```

I'm actually unsure which Java SDK is being used on this machine, since I'm not a Java developer, and javac is installed under /usr/bin/ while I thought it would be installed by Homebrew instead. 

```
$ which javac
/usr/bin/javac
$ javac -version
javac 1.8.0_282
```

Changing `h2` to `h4` on the line in question fixes the issue.

--- a/build.gradle
+++ b/build.gradle
@@ -229,6 +229,14 @@ project.afterEvaluate {
     }
 }
 
+// Disable HTML doclint to work around heading tag sequence validation
+// inconsistencies between JDK15 and earlier Java versions.
+allprojects {
+    tasks.withType(Javadoc) {
+        options.addStringOption('Xdoclint:-html', '-quiet')
+    }
+}
+
 /*
  * Sonatype Staging Finalization
  * ====================================================

